src/fetchServices.ts

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env node
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {google} from 'googleapis';
+const spreadsheetId = '14JYzl_8W3HyD0c1jYDXQJA2sVWBZiMoxT8Sp57xGWvk';
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.list endpoint
+ * @return {string[]} an array of hostnames ('foo.googleapis.com') for all available services
+ */
+export async function getAllServiceNames(): Promise<string[]> {
+  try {
+    const auth = await google.auth.getClient({
+      scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+    });
+    const client = google.servicemanagement({
+      version: 'v1',
+      auth: auth,
+    });
+    const serviceNames: string[] = [];
+    return await listServices(client, serviceNames);
+  } catch (e) {
+    console.log(e);
+    throw e;
+  }
+}
+
+export async function listServices(
+  client: any,
+  serviceNames: string[],
+  nextPageToken = ''
+): Promise<any> {
+  const res = await client.services.list(
+    {
+      pageSize: 100,
+      pageToken: nextPageToken,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+
+  if (res.data.services) {
+    res.data.services.map((item: any) => {
+      serviceNames.push(item.serviceName);
+    });
+  }
+
+  if (res.data.nextPageToken) {
+    await listServices(client, serviceNames, res.data.nextPageToken);
+  }
+  return serviceNames;
+}
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.getConfig endpoint
+ * @param {string} serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return {Object} the service config of the passed hostname
+ */
+export async function getServiceConfig(serviceName: string): Promise<any> {
+  try {
+    const auth = await google.auth.getClient({
+      scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+    });
+    const servicemanagement = google.servicemanagement({
+      version: 'v1',
+      auth: auth,
+    });
+    const res = await servicemanagement.services.getConfig(
+      {
+        serviceName: serviceName,
+      },
+      {
+        retryConfig: {
+          retry: 10,
+        },
+      }
+    );
+    return res.data;
+  } catch (e) {
+    console.log(e);
+    throw e;
+  }
+}
+
+/**
+ * Get service configs for all available services.
+ * @return {Object}
+ */
+export async function getAllServiceConfigs(): Promise<any> {
```

Comment:
Fun fact:  if you remove the `: Promise<any>` annotation all together, the compiler will more accurately infer the return type :P